crates/kornia-imgproc/src/flip.rs

```diff
@@ -97,31 +100,44 @@ where
 /// )
 /// .unwrap();
 ///
-/// let flipped: Image<f32, 3> = vertical_flip(&image).unwrap();
+/// let mut flipped = Image::<f32, 3>::from_size_val(
+///     ImageSize {
+///         width: 2,
+///         height: 3,
+///     },
+///     0.0
+/// )
+/// .unwrap();
+///
+/// vertical_flip(&image, &mut flipped).unwrap();
 ///
-/// assert_eq!(flipped.size().width, 2);
-/// assert_eq!(flipped.size().height, 3);
 /// ```
-pub fn vertical_flip<T, const C: usize>(src: &Image<T, C>) -> Result<Image<T, C>, ImageError>
+pub fn vertical_flip<T, const C: usize>(src: &Image<T, C>, dst: &mut Image<T, C>) -> Result<(), ImageError>
 where
     T: SafeTensorType,
 {
-    let mut dst = src.clone();
+    if src.size() != dst.size() {
+        return Err(ImageError::InvalidImageSize(
+            src.cols(),
+            src.rows(),
+            dst.cols(),
+            dst.rows(),
+        ));
+    }
 
     // TODO: improve this implementation
```

Comment:
I retained this for your review. (I'm very new to rayon so I'm not sure if the below logic has any inefficiencies).